Examples/UIComponentExample/Examples/Complex layout/ComplexLayoutViewController.swift

```diff
@@ -0,0 +1,219 @@
+//
+
+import UIComponent
+import UIKit.UIButton
+
+class ComplexLayoutViewController: ComponentViewController {
+  
+  var showWeather: Bool = false
+  
+  static let originalIds = [Cell.Context(fillColor: colors.randomElement()!, id: "1"),
+                            Cell.Context(fillColor: colors.randomElement()!, id: "2"),
+                            Cell.Context(fillColor: colors.randomElement()!, id: "3"),
+                            Cell.Context(fillColor: colors.randomElement()!, id: "4"),
+                            Cell.Context(fillColor: colors.randomElement()!, id: "5")]
+  static let colors: [UIColor] = [.systemRed, .systemBlue, .systemPink, .systemTeal, .systemGray, .systemFill, .systemGreen, .systemGreen, .systemYellow, .systemPurple, .systemOrange]
+  
+  var ids = originalIds {
+    didSet {
+      reloadComponent()
+    }
+  }
+  
+  var tags: [(String, UIColor)] = [("SwiftUI", .systemRed),
+                                   ("Hello", .systemOrange),
+                                   ("UIComponent", .systemPink),
+                                   ("Vue", .systemGreen),
+                                   ("Flex Layout", .systemTeal)] {
+    didSet {
+      reloadComponent()
+    }
+  }
+  
+  var expand = false
+  
+  lazy var waterfallData: [(CGSize, UIColor)] = {
+    var sizes = [(CGSize, UIColor)]()
+    for _ in 1 ... 30 {
+      sizes.append((CGSize(width: Int(arc4random_uniform(300 - 100)) + 100, height: Int(arc4random_uniform(300 - 100)) + 100), ComplexLayoutViewController.colors.randomElement()!))
+    }
+    return sizes
+  }()
+  
+  lazy var resetButton: UIButton = {
+    let button = UIButton(type: .system)
+    button.setTitle("Reset", for: .normal)
+    button.addTarget(self, action: #selector(self.resetIds), for: .touchUpInside)
+    return button
+  }()
+  
+  lazy var shuffledBuuton: UIButton = {
+    let button = UIButton(type: .system)
+    button.setTitle("Shuffled", for: .normal)
+    button.addTarget(self, action: #selector(self.resetIds), for: .touchUpInside)
+    return button
+  }()
+  
+  
+  override var component: Component {
+    VStack(spacing: 20) {
+      Text("Complex layouts", font: .boldSystemFont(ofSize: 20)).size(width: .fill).id("label")
+      VStack(spacing: 10) {
+        UserProfile(avatar: "https://unsplash.com/photos/Yn0l7uwBrpw/download?force=true&w=640",
+                    userName: "Jack",
+                    introduce: "This is simply amazing",
+                    gender: "🦄",
+                    showWeather: showWeather)
+        Text(showWeather ? "Hide weather" : "Show weather").textColor(.systemBlue).tappableView { [weak self] in
+          guard let self = self else { return }
+          self.showWeather = !self.showWeather
+          self.reloadComponent()
+        }.id("2")
+      }
+      
+      IntroductionCard(expand: expand) { [weak self] value in
+        self?.expand = !value
+        self?.reloadComponent()
```

Comment:
instead of calling reloadComponent here. let's make it consistent and call it in `didSet`

Reply:
✅